shellpython/preprocessor.py

```diff
@@ -76,12 +76,18 @@ def _get_username():
 
     :return: The name of current user
     """
-    return getpass.getuser()
+    try:
+        n = getpass.getuser()
+        return n
+    except:
+    #returns temporary name. eg: temp1232433
+        from time import time
```

Comment:
This path is the path to shellpy cache. That is the place where processed files are located. It must be stable. Otherwise if a script executes once in a second on some server it will generate huge amount of directories in /tmp/ directory if you use temp'+str(int(time)) to generate the path